src/backend/distributed/utils/directory.c

```diff
@@ -0,0 +1,215 @@
+/*-------------------------------------------------------------------------
+ *
+ * directory.c
```

Comment:
I guess all the functions here copy & paste as well?

Reply:
yes